crates/hyperswitch_domain_models/src/payments.rs

```diff
@@ -82,70 +82,179 @@ impl PaymentIntent {
     }
 
     #[cfg(all(feature = "v2", feature = "payment_v2",))]
-    pub fn get_id(&self) -> &id_type::PaymentId {
+    pub fn get_id(&self) -> &id_type::PaymentGlobalId {
         &self.id
     }
 }
 
+#[cfg(feature = "v2")]
+#[derive(Clone, Debug, PartialEq, serde::Serialize)]
+pub enum TaxCalculationOverride {
+    /// Skip calling the external tax provider
+    Skip,
+    /// Calculate tax by calling the external tax provider
+    Calculate,
+}
+
+#[cfg(feature = "v2")]
+#[derive(Clone, Debug, PartialEq, serde::Serialize)]
+pub enum SurchargeCalculationOverride {
+    /// Skip calculating surcharge
+    Skip,
+    /// Calculate surcharge
+    Calculate,
+}
+
+#[cfg(feature = "v2")]
+impl From<Option<bool>> for TaxCalculationOverride {
+    fn from(value: Option<bool>) -> Self {
+        match value {
+            Some(true) => TaxCalculationOverride::Calculate,
+            _ => TaxCalculationOverride::Skip,
+        }
+    }
+}
```

Comment:
I don't think it is a good idea to map None and Some(false) to TaxCalculationOverride::Skip.
During conversion from domain model to diesel model, we will end up populating surcharge_applicable field with Some(false) even if it was None in DB.

Reply:
In KV store

Reply:
yes that is expected. We should always have a value, either `true` or `false` in the database

Reply:
Since `surcharge_applicable` in `payment_intent` is an inferred field. It should be in `payment_attempt`.

`surcharge_applicable` is set to true during payment_method_list call. If any surcharge is calculated through the surcharge rules, we'll set `surcharge_applicable` to `true`. 
Here you have considered it as flag to skip surcharge calculation. 
The decision to skip surcharge calculation is made through `surcharge_amount` field in payment_intent. If the merchant sends some `surcharge_amount` through payments create call, we'll will skip surcharge calculation and send whatever surcharge sent by the merchant to the connector.

Reply:
All the fields in `PaymentIntent` are about the intent of the merchant. In this case we cannot use this field for our internal use. This behaviour will need to change in v2. This field will only tell about whether merchant wants us to calculate the surcharge or not.